### ISSUE TITLE
Does not read emojis in aria-label

### ISSUE THREAD

**Issue by Leon-Lj** [nvaccess/nvda#14296]
NVDA does not read emojis that are in `aria-lable`, it just skips them and reads the rest of the label.

### Steps to reproduce:
1. Create a html file with content:
```
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8" />
    <title>NVDA Emoji Test</title>
  </head>
  <body>
    <main>
	<div role="math" aria-label="🌻+2=6^{2}">
	  🌻+2=6<sup>2
	</div>
    </main>
  </body>
</html>
```
2. Open the html file in Chrome
3. Enter the main region
4. Use arrow keys to read the math expression
### Actual behavior:

`  +2=6^{2}`

**Comment by michaelDCurran** [comment]
Confirmed.
I reproduced this with my NvDA log level set to input / output. 
After speaking the main landmark, it then tried to speak two 0xfffd characters. which I believe are Unicode placeholder characters.
I think this may be a bug in NVDA where we have deliberately converted unknown / private use unicode characters into safe characters before explorting them via attributes in our internal xml representation of the content as it gets sent from the browser process to NVDA.
this is most likely then NVDA being too broad with its checks. There should certainly be a valid way of storing imoji unicode characters in an xml attribute string.
For anyone looking to fix this:
Please refer to `appendCharToXML` in `nvdaHelper/common/xml.h`
